lib/rugged/repository.rb

```diff
@@ -210,6 +210,17 @@ def fetch(remote_or_url, *args)
       remote_or_url.fetch(*args)
     end
 
+    # Get the tree at a path for a specific revision.
+    #
+    # revision - The String SHA1.
+    # path     - The String file path.
+    #
+    # Returns a String.
+    def tree_at(revision, path)
+      tree = Rugged::Commit.lookup(self, revision).tree
+      tree.path(path)[:oid]
```

Comment:
The way this is implemented right now, `#tree_at` would return the tree's oid, not the actual `Rugged::Tree` object. This is different from how `#blob_at` works.

Would you mind fixing this up and adding some test cases?